.github/workflows/extraction_test.yml

```diff
@@ -0,0 +1,151 @@
+        fail-fast: true
+        commit-filter: '[skip ci]'
+    - name: Install dependencies
+      run: |
+        # Use a specific version of Python (rather than the system-wide one).
+        module load python/${PYTHON_VERSION}
+        python -m venv .runner_venv
+        . .runner_venv/bin/activate
+        python -m pip install --upgrade pip
+        # Uncomment the below to use the submodule version of fparser rather
+        # than the latest release from pypi.
+        # pip install external/fparser
+        pip install .[test]
+        # Install BAF, and its version of Fab:
+        # ------------------------------------
+        git clone -b main --recurse-submodules https://github.com/MetOffice/lfric-baf.git
+        # First install Fab:
+        cd lfric-baf/fab
+        pip install .
+
+    # Compile GungHo with extraction
+    - name: GungHo Extraction Compilation
+      run:  |
+        # Set up environment
+        source /apps/spack/psyclone-spack/spack-repo/share/spack/setup-env.sh
+        spack load lfric-build-environment%gcc@14
+        source .runner_venv/bin/activate
+        export LFRIC_DIR=${HOME}/LFRic/lfric_apps_${LFRIC_APPS_REV}
+        export PSYCLONE_CONFIG_FILE=${PSYCLONE_LFRIC_DIR}/KGOs/lfric_psyclone.cfg
+
+        # Set the directory names for the extraction runs:
+        # -----------------------------------------------
+        export RUNDIR=${HOME}/LFRic/extraction_test
+        export LFRIC_CORE=${RUNDIR}/lfric_core_${LFRIC_CORE_REV}
+        export LFRIC_APPS=${RUNDIR}/lfric_apps_${LFRIC_APPS_REV}
+
+        # Copy the lfric source files (BAF copies files into the source tree)
+        # -------------------------------------------------------------------
+        rm -rf ${RUNDIR}
+        mkdir -p ${LFRIC_CORE} ${LFRIC_APPS}
+        cp -r ${HOME}/LFRic/lfric_core_${LFRIC_CORE_REV}/ ${RUNDIR}
+        cp -r ${HOME}/LFRic/lfric_apps_${LFRIC_APPS_REV}/ ${RUNDIR}
+
+        # Install the BAF scripts into the lfric directories:
+        # ---------------------------------------------------
+        cd lfric-baf
+        ./install.sh ${LFRIC_CORE} ${LFRIC_APPS}
+
+        # Build GungHo with the BAF extraction script
+        # -------------------------------------------
+        cd ${LFRIC_APPS}/applications/gungho_model
+        mkdir -p $PWD/fab_workspace
+        export FAB_WORKSPACE=$PWD/fab_workspace
+        # Fab expects that all compilers are in PATH, and
+        # does not support CC/FC etc with a full path. Since
+        # the spack environment does not add all required paths,
+        # we need to add the required paths manuall, and then
+        # unset the environment variables (which would otherwise
+        # be used as default)
+        export PATH="$(dirname $CC):$(dirname $LDMPI):/apps/spack/psyclone-spack/spack-repo/opt/spack/linux-centos9-skylake_avx512/gcc-14.1.0/rose-picker-2-mmt2v3hqjwoqgkyf4f2ijlzlsy7nubnp/bin:$PATH"
```

Comment:
Can the hardcoded rose-picker path with replaces with a "spack location -i rose-picker" command?